x-pack/plugins/infra/public/components/source_configuration/source_configuration_flyout.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import React from 'react';
+
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n/react';
+import { WithSource } from '../../containers/with_source';
+import { FieldsConfigurationPanel } from './fields_configuration_panel';
+import { IndicesConfigurationPanel } from './indices_configuration_panel';
+import { NameConfigurationPanel } from './name_configuration_panel';
+import { WithSourceConfigurationFlyoutState } from './source_configuration_flyout_state';
+import { WithSourceConfigurationFormState } from './source_configuration_form_state';
+
+const noop = () => undefined;
+
+interface SourceConfigurationFlyoutProps {
+  intl: InjectedIntl;
+}
+
+export const SourceConfigurationFlyout = injectI18n(({ intl }: SourceConfigurationFlyoutProps) => (
+  <WithSourceConfigurationFlyoutState>
+    {({ disable: close, value: isVisible }) =>
+      isVisible ? (
+        <WithSource>
+          {({ create, configuration, exists, isLoading, update }) =>
+            configuration ? (
+              <WithSourceConfigurationFormState
+                initialState={{
```

Comment:
I'm wondering if it is better to let the container handle its `initialState` internally, instead of passing it in.
Do you think we will use `WithSourceConfigurationFormState` with different initial states, ever?

Reply:
Result of the offline discussion: I misunderstood what `initialState` contains here, it is the state of the form before the user made any changes, and not the fallback to the initial / default values of a source configuration.

A less generic name would maybe be helpful, if you can find a good one.

Reply:
Maybe `initialFormState` clarifies that the "initial" only refers to the state of the form and not the default configuration?

Reply:
Yes, thank you!